### PR TITLE
feat: improve sorting and filtering

### PR DESCRIPTION
**Pull request by d-loose**
UDENG-797

* Generalizes the sorting functionality from the search page so that it can be reused elsewhere
* Adds sorting and filtering to the manage page

[Screencast from 2023-08-18 17-29-35.webm](https://github.com/ubuntu/app-store/assets/113362648/9d30c8eb-65ab-4b14-b70c-ef9e89984a21)

### CODE REVIEW

File: lib/src/manage/local_snap_providers.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+final localSnapsProvider = Provider.autoDispose(
+  (ref) => ref
+      .watch(manageModelProvider.select((m) => m.nonRefreshableSnaps))
+      .where((snap) => snap.titleOrName
+          .toLowerCase()
+          .contains(ref.watch(localSnapFilterProvider).toLowerCase()))
+      .where((snap) =>
+          ref.watch(showLocalSystemAppsProvider) || snap.apps.isNotEmpty)
+      .sortedSnaps(ref.watch(localSnapSortOrderProvider)),
+);
```
Comment (CarlosNihelton) [inline lib/src/manage/local_snap_providers.dart:19]: This seems hard to read at first, but the proxying of providers it offers is awesome!

File: lib/src/manage/manage_page.dart [hunk 2]
```diff
@@ -106,13 +108,66 @@ class _ManageView extends ConsumerWidget {
...
+                    final sortOrder = ref.watch(localSnapSortOrderProvider);
+                    return MenuButtonBuilder<SnapSortOrder>(
+                      values: const [
+                        SnapSortOrder.alphabeticalAsc,
+                        SnapSortOrder.alphabeticalDesc,
+                        SnapSortOrder.installedDateAsc,
+                        SnapSortOrder.installedDateDesc,
+                        SnapSortOrder.installedSizeAsc,
+                        SnapSortOrder.installedSizeDesc,
+                      ],
```
Comment (CarlosNihelton) [inline lib/src/manage/manage_page.dart:143]: Did we consider the download size also as an ordering criteria?

**Review by CarlosNihelton — APPROVED** [review]
LGTM 👍🏽

File: lib/src/manage/manage_page.dart (hunk 2 above)
Comment (d-loose) [reply lib/src/manage/manage_page.dart:143]: No, local snaps don't have a download size.

File: lib/src/manage/local_snap_providers.dart (hunk 1 above)
Comment (d-loose) [reply lib/src/manage/local_snap_providers.dart:19]: Yes, it works really well for this purpose :)

**Pull request merged**